### DISCUSSION TITLE
What is the difference between slice and splice?

### DISCUSSION THREAD

**Discussion by montasim** [STRINGLABITSOLUTIONS/Bhojon#345] [Q&A] [answered]
In short

**Comment by d4tdev** [comment, marked as answer]
`slice` returns a piece of the array but it doesn’t affect the original array. `splice` changes the original array by removing, replacing, or adding values and returns the affected values.

When you use each one is up to you. If you have to alter the content of the array, `splice` is the way to go, but if you just want to obtain a subarray, `slice` should be your choice

**Comment by Preetiraj3697** [comment]
Slice is used to get a new array from the original array whereas the splice is used to add/remove items in the original array. The changes are not reflected in the original array in the case of slice and in the splice, the changes are reflected in the original array.
